Upgrade Solution to 11.0.4.Final (#4)
infinispan/infinispan-server-tutorial: README.adoc

To marshall the annotated `LocationWeather` class, Infinispan requires a Protobuf schema. You can either provide a Protobuf descriptor file or create a descriptor file from the annotations you added to the POJO.

In `LocationWeatherMarshallingContext`, you add the schema to the Protobuf cache in Infinispan and then build a Protobuf using the `@AutoProtoSchemaBuilder` method.

.org.infinispan.tutorial.db.LocationWeatherMarshallingContext
[source,java]
----
@AutoProtoSchemaBuilder(
includeClasses = {
LocationWeather.class
},
schemaFileName = "weather.proto",
schemaFilePath = "proto/",
schemaPackageName = "org.infinispan.tutorial.data")
public interface LocationWeatherSchema extends SerializationContextInitializer {
}
----

[TIP]
====

> Run `mvn clean package` in the command line or build the project from the editor to generate the `LocationWeatherSchemaImpl` class.

Reviewer: ```suggestion
Run `mvn clean package` from the command line or build the project in your IDE to generate the `LocationWeatherSchemaImpl` class.
```

----

   @RegisterExtension
   static InfinispanServerExtension infinispanServerExtension = InfinispanServerExtensionBuilder.server(); // <1>

   @Test
   public void loadLocationTemperature() {
      DataSourceConnector dataSourceConnector = new DataSourceConnector(createRemoteCacheManager());
      TemperatureLoader temperatureLoader = new TemperatureLoader(dataSourceConnector);
      Float temperatureLoaderForLocation = temperatureLoader.getForLocation(WeatherLoader.LOCATIONS[0]);
      assertNotNull(temperatureLoaderForLocation);
   }

   // <2>
   private RemoteCacheManager createRemoteCacheManager() {
      RemoteCacheManager remoteCacheManager = infinispanServerExtension.hotrod().createRemoteCacheManager();
      SerializationContext serCtx = MarshallerUtil.getSerializationContext(remoteCacheManager);
      LocationWeatherSchema schema = new LocationWeatherSchemaImpl();
      schema.registerSchema(serCtx);
      schema.registerMarshallers(serCtx);
      return remoteCacheManager;
   }
----

> <1> Registers the Junit 5 Infinispan Server Extension

Reviewer: ```suggestion
<1> Registers the Junit 5 Infinispan Server Extension.
```

> <2> Configures the serialization context in the tests

Reviewer: ```suggestion
<2> Adds a serialization context for the tests.
```
